### PR TITLE
Clean up duplicate JSON.parse calls into re-usable helper

### PR DESCRIPTION
**Pull request by darokel**
### Context

- I noticed a lot of duplicated calls to `JSON.parse` for parsing the response body in test files

### Changes proposed in this pull request

- Standardise helper method for parsing JSON responses in test files to avoid duplicating calls. 
- Instead of `JSON.parse` you would just do `json_response` which would also memoize the results.

### Guidance to review

- See the new helper method

### CODE REVIEW

File: spec/support/request_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module RequestHelpers
+  def json_response
+    @json_response ||= JSON.parse(response.body)
```
Comment (misaka) [inline spec/support/request_helpers.rb:3]: May be more prudent to use `@_json_response` for a memoised value here.

Comment (darokel) [reply spec/support/request_helpers.rb:3]: @misaka Currently this raises a rubocop violation https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/MemoizedInstanceVariableName. We would need to change `EnforcedStyleForLeadingUnderscores: disallowed (default)` to something like `required` or `optional`, thoughts?

Comment (misaka) [reply spec/support/request_helpers.rb:3]: As discussed, I'd be happy with `optional` here. I personally prefer the underscore here, but I understand that it's not always required.

Comment (darokel) [reply spec/support/request_helpers.rb:3]: done https://github.com/DFE-Digital/teacher-training-api/pull/1528/commits/991061d39e4caf7deaf073f8e1f29c929a2a089a

**Pull request merged**